crates/runtime/src/memory.rs

```diff
@@ -484,12 +486,18 @@ impl SharedMemory {
     pub fn vmmemory_ptr(&self) -> *const VMMemoryDefinition {
         &self.0.read().unwrap().def.0 as *const _
     }
+
+    /// Return a reference to the shared memory's [ParkingSpot].
+    pub fn parking_spot(&self) -> Arc<ParkingSpot> {
```

Comment:
If this returned `&Arc<ParkingSpot>` what other errors pop up?

Reply:
```
error[E0515]: cannot return value referencing temporary value
   --> crates/runtime/src/memory.rs:492:9
    |
492 |         &self.0.read().unwrap().spot
    |         ^----------------------^^^^^
    |         ||
    |         |temporary value created here
    |         returns a value referencing data owned by the current function

```

Reply:
nevertheless, by adding `Shared::Memory::atomic_*` methods, I got  rid of it.